playbooks/roles/mongo_3_2/templates/backup-mongo.sh.j2

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+# ref https://tasks.opencraft.com/browse/SE-1669
+# Script to perform a point-in-time dump of the local mongodb database using
+# mongodump.
+# includes locking (prevent this script from running multiple times in
+# parallel), creating a snapshot of the volume used to backup to, and snitch monitoring
+
+# exit by default on failure
+set -e
+
+# vars
+ARCHIVE_NAME="mongo-backup-$(date --iso-8601=minutes --utc).gz"
+VOLUME="{{ MONGO_BACKUP_EBS_VOLUME_DEVICE }}" # eg. /dev/svdk
+VOLUME_ID="{{ MONGO_BACKUP_EBS_VOLUME_ID }}" # eg. vol-123456
+VOLUME_MOUNT_PATH="{{ MONGO_BACKUP_VOLUME_MOUNT_PATH }}" # eg. vol-123456
+SNITCH_URL="{{ MONGO_BACKUP_SNITCH_URL }}"
+MONGO_USER="{{ COMMON_MONGO_READ_ONLY_USER }}"
+MONGO_PASS="{{ COMMON_MONGO_READ_ONLY_PASS }}"
+HOSTNAME="{{ MONGO_BACKUP_HOST }}"
+
+# TODO: is hostname what we want here?
+if [ "$HOSTNAME" = $(hostname) ]; then
```

Comment:
@pomegranited what's a safe way to refer to a particular instance? hostname?

Also note to self: the mongo server on which this backup is to run must be configured as a read-only/non-voting/non-public instance.

Reply:
Yep -- let me know if you need help with that?  It'll be in the `rs.config`.

Reply:
Hmm I just ran that command and it appears be a glorified method of getting the instance's ip address. It appears we are already hardcoding the mongo hosts as ip addresses, so wouldn't it be simpler to use `hostname` or `hostname -i`? Not sure what would be more stable.

Reply:
Hmm interesting.. maybe the node names default to IP if there's not a different one specified?  Not sure if it's possible to rename them with these roles.

The other thing that's implicitly verified by using the mongo mechanism is, we're checking that this node is still in the cluster.  I'm not fussed either way, but am inclined to use the same mechanism as the old script, in the hope that it'll make this more acceptable upstream?

Reply:
Yep fair enough. done :)